packages/core/src/browser/shell/tab-bars.ts

```diff
@@ -571,6 +602,14 @@ export class ScrollableTabBar extends TabBar<Widget> {
         if (!this.scrollBar) {
             this.scrollBar = this.scrollBarFactory();
         }
+        this.node.addEventListener('mouseenter', () => { this.isMouseOver = true; });
+        this.node.addEventListener('mouseleave', () => {
```

Comment:
This gives a slightly odd behavior if tabs are closed but we remain in the tabbar, perhaps we should recompute when a tab is created/closed?


https://user-images.githubusercontent.com/40359487/231261339-42eb81e8-5617-4fe9-b09c-7a9a78156d5c.mov

Reply:
The behavior is intentional in order to keep the position of the "close" icon stable as we repeatedly close tabs. It's inspired by what Chrome does. I would argue the utility in this case outweighs the visual oddity.